pkg/kubelet/cm/devicemanager/pod_devices.go

```diff
@@ -321,10 +328,70 @@ func (pdev *podDevices) deviceRunContainerOptions(podUID, contName string) *Devi
 			annotationsMap[k] = v
 			opts.Annotations = append(opts.Annotations, kubecontainer.Annotation{Name: k, Value: v})
 		}
+
+		// Updates for CDI devices. This also updates the annotations.
+		cdiDevices, cdiAnnotations := getCDIDeviceInfo(
+			podUID, resourceName,
+			resp.CdiDevices,
+			annotationsMap,
+			cdiDevicesMap,
+		)
+		opts.CDIDevices = append(opts.CDIDevices, cdiDevices...)
+		opts.Annotations = append(opts.Annotations, cdiAnnotations...)
 	}
 	return opts
 }
 
+// getCDIDeviceInfo returns CDI devices and annotations for a given pod.
+func getCDIDeviceInfo(podUID, resourceName string, responseCDIDevices []*pluginapi.CDIDevice, annotationsMap map[string]string, cdiDevicesMap map[string]struct{}) ([]kubecontainer.CDIDevice, []kubecontainer.Annotation) {
+	if !utilfeature.DefaultFeatureGate.Enabled(kubefeatures.DevicePluginCDIDevices) {
+		return nil, nil
+	}
```

Comment:
In general, we try and make the calls to the check the feature gate at the highest level possible, rather than buried within function calls that "do nothing" if the feature gate is disabled. In this case, that would mean pulling the feature gate up into pod_device.go and wrapping the following with it:

```
		// Updates for CDI devices. This also updates the annotations.
		cdiDevices, cdiAnnotations := getCDIDeviceInfo(
			podUID, resourceName,
			resp.CdiDevices,
			annotationsMap,
			cdiDevicesMap,
		)
		opts.CDIDevices = append(opts.CDIDevices, cdiDevices...)
```

Reply:
OK. I will update.